src/gtMerger/matchAndMerge.py

```diff
@@ -164,6 +182,36 @@ def preprocessGtData(gt_data):
     return clean_gt_data
 
 
+def merge(gt_data, oba_data):
+    """
+    Merge gt_data dataframe and oba_data dataframe using the nearest value between columns 'gt_data.ClosestTime' and
+    'oba_data.Activity Start Date and Time* (UTC)'. Before merging, the data is grouped by 'GT_Collector' on gt_data and
+    each row on gt_data will be paired with one or none of the rows on oba_data grouped by userId.
+    :param gt_data: dataframe with preprocessed data from ground truth XLSX data file
+    :param oba_data: dataframe with preprocessed data from OBA firebase export CSV data file
+    :return: dataframe with the merged data.
+    """
+    list_collectors = gt_data['GT_Collector'].unique()
+    list_oba_users = oba_data['User ID'].unique()
+
+    merged_df = pd.DataFrame()
+    for collector in list_collectors:
+        print("Merging data for collector ", collector)
+        # Create dataframe for a collector on list_collectors
+        gt_data_collector = gt_data[gt_data["GT_Collector"] == collector]
+        for oba_user in list_oba_users:
+            print("\t Oba user", oba_user)
```

Comment:
Does this include the OBA data user ID? If so, let's print an simple counting integer instead, or truncate the user ID to the last 4 digits if you need to see the ID. I'd prefer not to leak user IDs in log output.